functions/src/SQR.functions.js

```diff
@@ -0,0 +1,234 @@
+
+const functions = require('firebase-functions');
+const admin = require("firebase-admin");
+
+const bucket = admin.storage().bucket();
+const db = admin.database();
+const helpers = require('./../helpers');
+
+
+
+const SQR_Helper = require('./SQR_Helper');
+/////////////////////////////////////////////////
+//          OnNewAllotment (DB create and update Trigger)
+//      1. Mark the files in the database --> { status: "Given" }
+//              Function --> UpdateFilesOnNewAllotment
+//
+//      2. Send an email to the devotee to notify them of the new allotments
+//              Function --> sendEmailOnFileAllotment
+/////////////////////////////////////////////////
+
+exports.updateFilesOnNewAllotment = functions.database.ref('/sqr/allotments/{allotment_id}')
+    .onCreate((snapshot, context) => {
+        return SQR_Helper.UpdateFilesOnNewAllotment(
+                snapshot, 
+                db,
+                helpers.updateFile
+            );
+});
+
+exports.sendEmailOnFileAllotment = functions.database.ref('/sqr/allotments/{allotment_id}')
+    .onUpdate((change, context) => {
+        const old = change.before.val();
+        const _new = change.after.val();        
+        let coordinatorConfig = functions.config().audioseva.coordinator;
+        
+        db.ref('/sqr/allotments').once('value')
+        .then(snapshot => {
+            const allotments = snapshot.val();
+            let alloted_num = 0, alloted_before = false;
+            for(let key in allotments) {
+                let allotment = allotments[key];
+                if(allotment.devotee)
+                    if(allotment.devotee.emailAddress)
+                        if(allotment.devotee.emailAddress === _new.devotee.emailAddress)
+                            alloted_before++;
+            }
+            if (alloted_before > 1)
+                alloted_before = true;
+            SQR_Helper.sendEmailOnFileAllotment(coordinatorConfig, old, _new, change.after, alloted_before, helpers.sendEmail)
```

Comment:
Particularly here, you will not send the same allotment email from other places, only from this one place. What is the use of extracting the code into helper?